### ISSUE TITLE
3rd Camera throws frame errors

### ISSUE THREAD

**Issue by SomebodySysop** [mrlt8/docker-wyze-bridge#3]
Added a 3rd camera, WyzeCam303.  Same settings as other two (as far as I can tell).  Still have to run wyze-to-rtmp manually, but now it won't connect at all, throwing these types of errors:

[h264 @ 0x55b333591e40] If you want to help, upload a sample of this file to ftp://upload.ffmpeg.org/incoming/ and contact the ffmpeg-devel mailing list. (ffmpeg-devel@ffmpeg.org)
[h264 @ 0x55b333591e40] no frame!
[h264 @ 0x55b333591e40] data partitioning is not implemented. Update your FFmpeg version to the newest one from Git. If the problem still occurs, it means that your file has a feature which has not been implemented.

[h264 @ 0x55b333591e40] If you want to help, upload a sample of this file to ftp://upload.ffmpeg.org/incoming/ and contact the ffmpeg-devel mailing list. (ffmpeg-devel@ffmpeg.org)
[h264 @ 0x55b333591e40] no frame!
[h264 @ 0x55b333590640] decoding for stream 0 failed
[h264 @ 0x55b333590640] Could not find codec parameters for stream 0 (Video: h264, none): unspecified size
Consider increasing the value for the 'analyzeduration' and 'probesize' options
Input #0, h264, from 'pipe:':
  Duration: N/A, bitrate: N/A
    Stream #0:0: Video: h264, none, 25 fps, 25 tbr, 1200k tbn, 50 tbc
[flv @ 0x55b333595b00] dimensions not set
Could not write header for output file #0 (incorrect codec parameters ?): Invalid argument
Stream mapping:
  Stream #0:0 -> #0:0 (copy)
    Last message repeated 1 times
Reconnect
Reconnect
Reconnect
Reconnect
Reconnect
Reconnect
/home/ron/wyze-to-rtmp/lib/python3.8/site-packages/wyzecam/iotc.py:367: UserWarning: Lost frame
  warnings.warn("Lost frame")
/home/ron/wyze-to-rtmp/lib/python3.8/site-packages/wyzecam/iotc.py:364: UserWarning: Received incomplete frame
  warnings.warn("Received incomplete frame")
Reconnect

Any suggestions for correction?

**Comment by SomebodySysop** [comment]
Figured it out.  On my home network, I've got 4 routers.  All are attached to the same network, but at different locations to allow for maximum connections.  The linux laptop and other 2 cameras are connected to one ssid.  The 3rd camera was connected to another ssid.  When I connected the 3rd camera to the same ssid as the others, no more frame errors.  Whew!

**Comment by noelhibbard** [comment]
Why don't you run them all on the same SSID but different channels? If your APs support 802.11k/v then clients should roam instantly as they move around. If they are on separate SSIDs then roaming will suck.... bad.

**Comment by SomebodySysop** [comment]
> Why don't you run them all on the same SSID but different channels? If your APs support 802.11k/v then clients should roam instantly as they move around. If they are on separate SSIDs then roaming will suck.... bad.

The reason I have multiple routers attached to same network is to extend coverage to different parts of the house.  I think I got the frame errors because that particular router is a 2.4 ghz while the router the rest of the cameras, and the linux box, are connected to is 5ghz.  Connecting all the cameras to the same router seemed to fix the problem.

**Comment by noelhibbard** [comment]
Right but they don't need separate SSIDs to cover the whole house. The proper way is to put them all on the same SSID otherwise roaming will not work very well as you move around your house. I see people commonly name them MyWiFi-24 and MyWifi-5 which is also improper (even though a lot of residential routers ship this way). The key is to make sure there is no channel overlap and enable 802.11k/v if your routers support it.

But anyways, I'm going off topic.

**Comment by SomebodySysop** [comment]
> Right but they don't need separate SSIDs to cover the whole house. The proper way is to put them all on the same SSID otherwise roaming will not work very well as you move around your house. I see people commonly name them MyWiFi-24 and MyWifi-5 which is also improper (even though a lot of residential routers ship this way). The key is to make sure there is no channel overlap and enable 802.11k/v if your routers support it.
> 
> But anyways, I'm going off topic.

That is a good idea.  Thanks!

**Comment by SomebodySysop** [comment]
OK, after all day working on this, climbing up and down that 20 foot ladder, I think I have some findings.  First off, with the wyze-bridge update from last night. all 3 of my cameras are running reasonably stable.

1. I used different access points for each camera, getting the camera as close to them as I could.
2. For one outside camera, I put the AP right up against the window.
3. I simply replaced the 301 camera with another, and connected it to another AP.  Its been running the smoothest since I started working with wyze-to-rtmp a few days ago.

So, in my case at least, it would appear that the bulk of connectivity problems were network related.  So, I would suggest those of you having similar problems first check your network connectivity with each camera.   It's not just how far the camera is from the router/ap, but also what is between the two, and whether one is low or high.    

The router that my 301 was connected to is my strongest in the house, and they were around 10 feet apart.  But, the router sits on the floor of a thick wood cabinet.  I connected another camera in the same position but to an AP which is weaker and further away, but sits up high and voila, no more connectivity problems.

So my conclusion for this issue is that either there was a problem with my camera 301, or there was a problem with how I had it connected.  In either case, issue resolved.  So far.
